src/main/java/org/petshelter/service/PetShelterServiceImpl.java

```diff
@@ -0,0 +1,43 @@
+package org.petshelter.service;
+
+
+import lombok.RequiredArgsConstructor;
+import org.petshelter.model.Pet;
+import org.petshelter.model.Shelter;
+
+import static java.lang.System.out;
+
+@RequiredArgsConstructor
+public class PetShelterServiceImpl implements PetShelterService {
+    private final Shelter shelter;
+
+    @Override
+    public void addPet(Pet newPet) {
+        try {
+            if (shelter.getPets() != null) {
```

Comment:
means that it is impossible to add a pet if the shelter is null? why don't to create new map and add the pet instead?